ComponentKitApplicationTests/CKStackLayoutComponentTests.mm

```diff
@@ -603,44 +583,321 @@ - (void)testCrossAxisStretchingOccursAfterStackAxisFlexing
+{
+  CKStackLayoutComponent *c =
+  [CKStackLayoutComponent
+   newWithView:kWhiteBackgroundView
+   size:{}
+   style:{
+     .direction = CKStackLayoutDirectionVertical
+   }
+   children:{
+     {
+       [CKComponent newWithView:{[UIView class], {{@selector(setBackgroundColor:), [UIColor redColor]}}} size:{50,150}],
+       .flexGrow = 1,
+     },
+     {
+       [CKComponent newWithView:{[UIView class], {{@selector(setBackgroundColor:), [UIColor blueColor]}}} size:{50,100}],
+       .flexShrink = 1,
+     },
+     {
+       [CKComponent newWithView:{[UIView class], {{@selector(setBackgroundColor:), [UIColor greenColor]}}} size:{50,150}],
+       .flexGrow = 0,
+     },
+     {
+       [CKComponent newWithView:{[UIView class], {{@selector(setBackgroundColor:), [UIColor yellowColor]}}} size:{50,200}],
+       .flexShrink = 3,
      },
    }];
+  // In this scenario a width of 400 results in a negative violation of 200.
+  // The first and third child components specify a flex grow factor of 1 and 0, respectively. They won't flex.
+  // The second child component specifies a flex grow factor of 1 and will flex by 50.
+  // The fourth child component specifies a flex grow factor of 3 and will flex by 150.
+  static CKSizeRange kSize = {{400, 400}, {400, 400}};
+  CKSnapshotVerifyComponent(c, kSize, nil);
+}
 
-  // A width of 400px results in a violation of 200px. This is distributed equally among each flexible component,
-  // causing both of them to be shrunk by 100px, resulting in widths of 300px, 100px, and 50px.
-  // In the W3 flexbox standard, flexible components are shrunk proportionate to their original sizes,
-  // resulting in widths of 180px, 100px, and 120px.
-  // This test verifies the current behavior--the snapshot contains widths 300px, 100px, and 50px.
-  static CKSizeRange kSize = {{400, 0}, {400, 150}};
+- (void)testNegativeViolationIsDistributedAmongFlexibleChildrenUntilOneShrinksToZeroWidth
+{
+  CKStackLayoutComponent *c =
+  [CKStackLayoutComponent
+   newWithView:kWhiteBackgroundView
+   size:{}
+   style:{
+     .direction = CKStackLayoutDirectionHorizontal
+   }
+   children:{
+     {
+       [CKComponent newWithView:{[UIView class], {{@selector(setBackgroundColor:), [UIColor redColor]}}} size:{300,50}],
+       .flexShrink = 1,
+     },
+     {
+       [CKComponent newWithView:{[UIView class], {{@selector(setBackgroundColor:), [UIColor blueColor]}}} size:{100,50}],
+       .flexShrink = 2,
+     },
+     {
+       [CKComponent newWithView:{[UIView class], {{@selector(setBackgroundColor:), [UIColor greenColor]}}} size:{200,50}],
+       .flexShrink = 1,
+     },
+   }];
+  // In this scenario a width of 400 results in a negative violation of 200.
+  // The first and third child components specify a flex shrink factor of 1 and will flex by 50.
+  // The second child component specifies a flex shrink factor of 2 and will flex by 100. It will have a width of 0.
```

Comment:
But wait, doesn't the spec require that the scaled flex shrink factor would not size the middle element to 0?

https://www.w3.org/TR/css-flexbox-1/#scaled-flex-shrink-factor